src/state/transactions/updater.tsx

```diff
@@ -24,7 +26,20 @@ export function toSerializableReceipt(receipt: TransactionReceipt): Serializable
   }
 }
 
+function getElapsedTime(): number | undefined {
+  // Only log this metric for the first swap of a session.
+  const ttsMarks = performance.getEntriesByName('tts-mark', 'mark')
+
+  if (ttsMarks.length > 0) {
+    return
+  }
+
+  performance.mark('tts-mark')
+  return performance.now() / 1000 // time-to-swap in seconds
```

Comment:
You're re-deriving the timing already captured in the mark. Instead, use the value from the mark returned by `performance.mark`.